x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/Lookup.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.plan.logical;
+
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.xpack.esql.io.stream.PlanStreamInput;
+import org.elasticsearch.xpack.esql.io.stream.PlanStreamOutput;
+import org.elasticsearch.xpack.esql.plan.logical.join.Join;
+import org.elasticsearch.xpack.esql.plan.logical.local.LocalRelation;
+import org.elasticsearch.xpack.ql.capabilities.Resolvables;
+import org.elasticsearch.xpack.ql.expression.Attribute;
+import org.elasticsearch.xpack.ql.expression.Expression;
+import org.elasticsearch.xpack.ql.expression.NamedExpression;
+import org.elasticsearch.xpack.ql.plan.logical.LogicalPlan;
+import org.elasticsearch.xpack.ql.plan.logical.UnaryPlan;
+import org.elasticsearch.xpack.ql.tree.NodeInfo;
+import org.elasticsearch.xpack.ql.tree.Source;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+
+import static org.elasticsearch.xpack.esql.expression.NamedExpressions.mergeOutputAttributes;
+
+/**
+ * Looks up values from the associated {@code tables}.
+ * The class is used only during the analysis phased, after which it is replaced by a {@code Join}.
+ */
+public class Lookup extends UnaryPlan {
+    private final Expression tableName;
+    private final List<NamedExpression> matchFields;
```

Comment:
Sorry, I made a mistake. I thought these were the expressions for the looked up columns, but this is what we match/join on. This is fine and follows the same pattern as Enrich.

What I was trying to suggest was having a `List<Alias>` that refers to the table columns and allows to rename the columns. Also, we may need a way to add only a subset of the columns - but both of these things can be done in the future.

(`NamedExpression` is more flexible - in case of `Enrich`, the enrich fields are encoded as such and, in practice, are either a `ReferenceAttribute` (enrich field without `WITH`) or `Alias` (enrich field with `WITH`).)